src/OrchardCore.Modules/OrchardCore.Users/Controllers/ExternalAuthenticationsController.cs

```diff
@@ -215,14 +219,16 @@ public async Task<IActionResult> ExternalLoginCallback(string returnUrl = null,
             }, ModelState.AddModelError);
 
             // If the registration was successful we can link the external provider and redirect the user.
-            if (iUser == null)
+            if (iUser == null || !ModelState.IsValid)
```

Comment:
When is it possible that we reach this while `!ModelState.IsValid`?

Reply:
The Register method can add error to the ModelState and make it invalid. Based on the implementation we return null if there was any error. But the consumer code should not depend on the implementation return value. So it better to add this check just in case the underlining implementation changes.

Reply:
OK!